### PR TITLE
alsa-state: Add FILESEXTREAPATHS for NXP specific and Mainline BSP

### PR DESCRIPTION
**Pull request by junzhuimx**
Current implement can only apply the setting under folder imx-generic-bsp. Add paths for the NXP and mainline specific setting.

### CODE REVIEW

**Review by kraj — APPROVED** [review]

File: recipes-bsp/alsa-state/alsa-state.bbappend [hunk 1]
```diff
@@ -1,4 +1,6 @@
 # Prepend path to override files from upstream recipe
 FILESEXTRAPATHS:prepend:imx-generic-bsp := "${THISDIR}/${PN}/imx-generic-bsp:"
+FILESEXTRAPATHS:prepend:imx-nxp-bsp := "${THISDIR}/${PN}/imx-nxp-bsp:"
+FILESEXTRAPATHS:prepend:imx-mainline-bsp := "${THISDIR}/${PN}/imx-mainline-bsp:"
```
Comment (thochstein) [inline recipes-bsp/alsa-state/alsa-state.bbappend:4]: I believe the way to fix this is to just drop the BSP overrides:

`FILESEXTRAPATHS:prepend := "${THISDIR}/${PN}:"

Comment (junzhuimx) [reply recipes-bsp/alsa-state/alsa-state.bbappend:4]: @thochstein You are right. It is much more simple to implement it. Let me update it

File: recipes-bsp/alsa-state/alsa-state.bbappend [hunk 2]
```diff
@@ -1,4 +1,4 @@
 # Prepend path to override files from upstream recipe
-FILESEXTRAPATHS:prepend:imx-generic-bsp := "${THISDIR}/${PN}/imx-generic-bsp:"
+FILESEXTRAPATHS:prepend := "${THISDIR}/${PN}:"
```
Comment (kraj) [inline recipes-bsp/alsa-state/alsa-state.bbappend:2]: What happens when you have other BSP layers in project ? Would it insert itself into those too ?

Comment (junzhuimx) [reply recipes-bsp/alsa-state/alsa-state.bbappend:2]: will not impact your case:
- if other BSP layers defines "imx-generic/nxp/mainline-bsp" in MACHINEOVERRIDES will pick related files under the special folder
- if no defined, they can add bbappend and add the same ILESEXTRAPATHS:prepend := "${THISDIR}/${PN}:" to pick up the version in their layers.
alsa-state
├── imx-generic-bsp
│   └── asound.state
├── imx-mainline-bsp
│   └── asound.conf
└── imx-nxp-bsp
    └── asound.conf

Comment (kraj) [reply recipes-bsp/alsa-state/alsa-state.bbappend:2]: That’s sounds good then. I am fine with this change

**Pull request merged**